shared/keystore/keystore.go

```diff
@@ -80,6 +81,38 @@ func (ks Store) GetKey(filename, password string) (*Key, error) {
 	return DecryptKey(keyjson, password)
 }
 
+// GetKeys from directory using the prefix to filter relevant files
+// and a decryption password.
+func (ks Store) GetKeys(directory, fileprefix, password string) ([]*Key, error) {
+	// Load the key from the keystore and decrypt its contents
+	// #nosec G304
+	files, err := ioutil.ReadDir(directory)
+	if err != nil {
+		return nil, err
+	}
+	keys := []*Key{}
+	i := 0
+	for _, f := range files {
+		n := f.Name()
+		filePath := filepath.Join(directory, n)
+		filePath = filepath.Clean(filePath)
+		if f.Mode().IsRegular() && strings.Contains(n, strings.TrimPrefix(fileprefix, "/")) {
+			keyjson, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				return nil, err
+			}
+			key, err := DecryptKey(keyjson, password)
+			if err != nil {
+				return nil, err
+			}
+			keys = append(keys, key)
```

Comment:
Can the keys be stored not in the array, but in the map, using the public key as the map key?
Or at least check the key for uniqueness.

Now, if there is the same key in different files (copying, backups, etc.), then a double vote is likely ...

Reply:
good observation.
Thanks

Reply:
Are you planning to address this comment? @shayzluf

Reply:
do we want to address this?

Reply:
done

Reply:
How is this done? It is still possible that the `keys` array could have duplicate keys

Reply:
I merged both pr into #2069 and addressed it there. sorry that I didn't notice that the comment is on an older pr